### ISSUE TITLE
Swift extension: support path for sourcekit-lsp

### ISSUE THREAD

**Issue by TeamPuzel** [zed-industries/zed#13302]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The Swift extension appears to be getting the lsp binary from Xcode, so when I'm using the latest Swift toolchain rather than the one from Xcode there's a mismatch as Zed uses an outdated language server.

```
2024-06-20T11:26:39+02:00 [INFO] starting language server. binary path: "/usr/bin/xcrun", working directory:
```

This results in missing types, attributes and keywords:

<img width="363" alt="image" src="https://github.com/zed-industries/zed/assets/94306330/a9b0d982-2e74-416e-8656-c946e0a80e36">

It would be an improvement if Zed respected which binary is actually in the path and only used Apple tools to get their binary as a fallback.

It also seems to ignore adding the path manually in my Zed configuration file

### Environment

Zed: v0.141.0 (Zed Preview)
OS: macOS 15.0.0
Memory: 32 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

N/A

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

N/A

**Comment by TeamPuzel** [comment]
> For reference the swift extension lives here: [zed-extensions/swift](https://github.com/zed-extensions/swift/).

Thanks, I was able to get it to check the path first 👍

**Comment by notpeter** [comment]
Wonderful, so this is ok to close?

For future reference / searchability can you paste any settings that got it working as expected?

**Comment by TeamPuzel** [comment]
> For future reference / searchability can you paste any settings that got it working as expected?

Ah, I didn't find any settings that worked, I instead added this to the source code

```rs
if let Some(path) = worktree.which("sourcekit-lsp") {
    return Ok(zed::Command {
        command: path,
        args: Default::default(),
        env: worktree.shell_env(),
    });
}
```

**Comment by maxdeviant** [comment]
This was done in https://github.com/zed-extensions/swift/pull/11 and is available in v0.3.0 of the Swift extension.
